server/services/core.js

```diff
@@ -10,6 +10,32 @@ const { isEmpty } = require('lodash');
 
 const { logMessage, getService, formatCache, mergeCache } = require('../utils');
 
+/**
+ * Add link x-default url to url bundles from strapi i18n plugin default locale.
+ *
+ * @param {object} config - The config object.
+ * @param {object} links - The language links.
+ *
+ * @returns {object | undefined} The default language link.
+ */
+const getDefaultLanguageLink = async (config, links) => {
+  let defaultLink;
+  const { getDefaultLocale } = strapi.plugin('i18n').service('locales');
+  const defaultLocale = await getDefaultLocale();
+
+  if (config.defaultLanguageUrlType === 'default-locale') {
+    // find url with default locale in generated bundle
```

Comment:
We only have to fetch the defaultLocale if the setting is actually set to 'default-locale'. Saves on some unnecessary queries.
```suggestion

  if (config.defaultLanguageUrlType === 'default-locale') {
    const { getDefaultLocale } = strapi.plugin('i18n').service('locales');
    const defaultLocale = await getDefaultLocale();
  
    // find url with default locale in generated bundle
```